### PR TITLE
[flutter_tools] reland: remove globals from android device/testing

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Re-land refactor of android_devices.dart and testing. Fixed because I forgot to provide a timeout configuration since it is not on globals.

### CODE REVIEW

File: packages/flutter_tools/lib/src/android/android_device_discovery.dart [hunk 1]
```diff
@@ -96,6 +106,12 @@ class AndroidDevices extends PollingDeviceDiscovery {
     String text, {
     List<AndroidDevice> devices,
     List<String> diagnostics,
+    AndroidSdk androidSdk,
+    FileSystem fileSystem,
+    Logger logger,
+    Platform platform,
+    ProcessManager processManager,
+    @required TimeoutConfiguration timeoutConfiguration,
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/android/android_device_discovery.dart:114]: I forgot to make this required, and it was null - causing the failure.

Comment (jmagman) [reply packages/flutter_tools/lib/src/android/android_device_discovery.dart:114]: How did that pass the Cirrus tests the first time?

**Review by jmagman — APPROVED** [review]
LGTM

**Review by christopherfujino — APPROVED** [review]
LGTM

File: packages/flutter_tools/lib/src/android/android_device_discovery.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/android/android_device_discovery.dart:114]: I guess no integration test hit android device discovery for real.

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/android/android_device_discovery.dart:114]: FWIW I did test the change initially, but not after I updated device discovery 🤦

**Pull request merged**